### ISSUE TITLE
Nuke doom-modeline from orbit

### ISSUE THREAD

**Issue by Cerebus** [SystemCrafters/crafted-emacs#269]
Doom-modeline/stable incompatible with eglot/stable (again!):

```
error in process filter: doom-modeline-update-eglot: Symbol’s function definition is void: eglot--spinner
error in process filter: Symbol’s function definition is void: eglot--spinner
```

Can't use doom-modeline/unstable b/c #248.

It's past time to remove this thing entirely.

**Comment by Cerebus** [comment]
doom-modeline/unstable has this change:

https://github.com/seagle0128/doom-modeline/commit/62b47fc8274c8cdff76f88fb6abcfe9d4aa602a3

But it remains unreleased to stable.

**Comment by jeffbowman** [comment]
It is frustrating that package has caused so much grief.  Personally I uninstalled the package and then added this to my `early-config.el` file - N.B. I prefer `package.el` so this works in that context.  Not sure how to replicate it with `straight.el`. 

```emacs-lisp
(customize-set-variable 'package-load-list
                        '((doom-modeline nil)
                          (tree-sitter-langs "0.12.8")
                          all))
```

Basically, this prevents `doom-modeline` from being installed or even activated by the package system if did have it installed.  Thus `autoload` functions aren't and all calls to the package result in errors related to void symbols and such.  (The line about `tree-sitter-langs` just locks the version at `0.12.8` after I installed it from a git commit with quelpa since tags past that version won't load - there is a fix, but no release yet).
